modules/dnn/src/dnn.cpp

```diff
@@ -638,6 +638,10 @@ struct Net::Impl
         fusion = true;
         preferableBackend = DNN_BACKEND_DEFAULT;
         preferableTarget = DNN_TARGET_CPU;
+#ifdef HAVE_OPENCL
+        if (ocl::useOpenCL())
+            preferableTarget = DNN_TARGET_OPENCL;
+#endif
```

Comment:
Until it's not tested well on different hardware, I suggest not to make OpenCL the preferable target. Let's have user to set it explicitly.